metadata page: add actuators (#3216)
PX4/PX4-user_guide: en/advanced/px4_metadata.md

# PX4 Metadata

PX4 uses and generates data that has associated human- and machine- readable metadata:

- [Parameters](../advanced_config/parameters.md) configure PX4 behaviour.
  - A parameter is represented by an ID string that maps to a value stored in PX4.
  - The associated metadata includes a description of the setting, its possible values, information about how the value might be presented (say for bitmasks).
- [Events](../concept/events_interface.md) provide notification of events, such as reasons for a failsafe, low battery warnings, end of calibration, and so on.
  - An event is represented by an id, and is sent with a log level and arguments.
  - The associated metadata includes a message, a description and a list of arguments (including their type) of each event.

> - [Actuators](../config/actuators.md) metadata contains information about supported vehicle geometries, a list of output drivers, and how to configure them. QGroundControl uses that information to dynamically build a configuration UI.

Reviewer: Just to my taste. 

```suggestion
- [Actuators](../config/actuators.md) metadata contains information about supported vehicle geometries, a list of output drivers, and how to configure them.
  QGroundControl uses that information to dynamically build a configuration UI.
```

# PX4 Metadata

PX4 uses and generates data that has associated human- and machine- readable metadata:

- [Parameters](../advanced_config/parameters.md) configure PX4 behaviour.
  - A parameter is represented by an ID string that maps to a value stored in PX4.
  - The associated metadata includes a description of the setting, its possible values, information about how the value might be presented (say for bitmasks).
- [Events](../concept/events_interface.md) provide notification of events, such as reasons for a failsafe, low battery warnings, end of calibration, and so on.
  - An event is represented by an id, and is sent with a log level and arguments.
  - The associated metadata includes a message, a description and a list of arguments (including their type) of each event.
- [Actuators](../config/actuators.md) metadata contains information about supported vehicle geometries, a list of output drivers, and how to configure them.
  QGroundControl uses that information to dynamically build a configuration UI.

> The metadata and metadata translations are shared with external systems, such as QGroundControl, allowing them to display information about parameters and events as a string in the user's own language.

Reviewer: ```suggestion
The metadata and metadata translations are shared with external systems, such as QGroundControl, allowing them to display information about parameters and events, and to configure vehicle geometry and actuator output mappings.
```

# PX4 Metadata

PX4 uses and generates data that has associated human- and machine- readable metadata:

- [Parameters](../advanced_config/parameters.md) configure PX4 behaviour.
  - A parameter is represented by an ID string that maps to a value stored in PX4.
  - The associated metadata includes a description of the setting, its possible values, information about how the value might be presented (say for bitmasks).
- [Events](../concept/events_interface.md) provide notification of events, such as reasons for a failsafe, low battery warnings, end of calibration, and so on.
  - An event is represented by an id, and is sent with a log level and arguments.
  - The associated metadata includes a message, a description and a list of arguments (including their type) of each event.

> - [Actuators](../config/actuators.md) metadata contains information about supported vehicle geometries, a list of output drivers, and how to configure them.
>   QGroundControl uses that information to dynamically build a configuration UI.
> 

Reviewer: This is not quite symmetric with the other headings above. I'll add this, but if you think it is wrong we can revert.

```suggestion
- [Actuators](../config/actuators.md) configuration customizes the specific geometry of the vehicle, assigns actuators and motors to flight controller outputs, and tests the actuator and motor response.
  - The metadata contains information about supported vehicle geometries, a list of output drivers, and how to configure them.
  - _QGroundControl_ uses that information to dynamically build a configuration UI.

```

> 

Reviewer: ```suggestion
- [Actuator Metadata](#actuator-metadata) (below)
```

The parameter XML file of the main branch is copied into the QGC source tree via CI and is used as a fallback in cases where no metadata is available via the component metadata protocol (this approach predates the existence of the component metadata protocol).
:::

> ### Actuator metadata

Reviewer: Convention
```suggestion
### Actuator Metadata
```
